go/cmd/dolt/commands/rebase.go

```diff
@@ -0,0 +1,369 @@
+func (cmd RebaseCmd) Description() string {
+	return rebaseDocs.ShortDesc
+}
+
+// EventType returns the type of the event to log
+func (cmd RebaseCmd) EventType() eventsapi.ClientEventType {
+	return eventsapi.ClientEventType_REBASE
+}
+
+func (cmd RebaseCmd) Docs() *cli.CommandDocumentation {
+	ap := cmd.ArgParser()
+	return cli.NewCommandDocumentation(rebaseDocs, ap)
+}
+
+func (cmd RebaseCmd) ArgParser() *argparser.ArgParser {
+	return cli.CreateRebaseArgParser()
+}
+
+func (cmd RebaseCmd) RequiresRepo() bool {
+	return false
+}
+
+// Exec executes the command
+func (cmd RebaseCmd) Exec(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv, cliCtx cli.CliContext) int {
+	ap := cmd.ArgParser()
+	help, usage := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, rebaseDocs, ap))
+	apr := cli.ParseArgsOrDie(ap, args, help)
+
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+	if closeFunc != nil {
+		defer closeFunc()
+	}
+
+	branchName, err := getActiveBranchName(sqlCtx, queryist)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+	rebaseBranchName := "dolt_rebase_" + branchName
+
+	query, err := constructInterpolatedDoltRebaseQuery(apr, rebaseBranchName)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+
+	rows, err := GetRowsForSql(queryist, sqlCtx, query)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+
+	status, err := getInt64ColAsInt64(rows[0][0])
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+	if status == 1 {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(errors.New("error: "+rows[0][1].(string))), usage)
+	}
+
+	message := rows[0][1].(string)
+	if strings.Contains(message, "interactive rebase completed") {
+		cli.Println("Successfully rebased and updated refs/heads/" + branchName + ".")
+	} else if strings.Contains(message, "interactive rebase aborted") {
+		cli.Println("Interactive rebase aborted.")
+	} else {
+		rebasePlan, err := getRebasePlan(cliCtx, sqlCtx, queryist, apr.Arg(0), branchName)
+		if err != nil {
+			return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+		}
+
+		if rebasePlan == nil || rebasePlan.Steps == nil {
+			rows, err := GetRowsForSql(queryist, sqlCtx, "CALL DOLT_REBASE('--abort');")
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			}
+			status, err := getInt64ColAsInt64(rows[0][0])
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			}
+			if status == 1 {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(errors.New("error: "+rows[0][1].(string))), usage)
+			}
+
+			cli.Println("Interactive rebase aborted.")
```

Comment:
I wasn't immediately sure why `rebasePlan` or `Steps` would be nil. I'll dig into the `getRebasePlan` function next. It seems like whatever case this is, we may want to display a different error message explaining that something went wrong and the rebase was aborted (i.e. explain why the rebase was aborted).

It may even be worth a comment on line 151 to explain when this case happens (if it's not clear from an updated cli.Println message).

Reply:
`rebasePlan.Steps` could be nil if the user deletes all uncommented out lines from the editor, which will abort the rebase. `getRebasePlan` will add nothing to the `Steps` if that's the case. The `rebasePlan == nil` check is mostly a safety precaution for checking `rebasePlan.Steps`. I can definitely add a comment here to give context for this check.